### PR TITLE
RUM-7112 chore: Add "RUM View Ended" telemetry to track TNS' and INV's quality

### PR DESCRIPTION
**Pull request by ncreated**
### What and why?

📦 🔭 Following the [spec](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/4534173727/RUM+View+Ended+Telemetry+-+Spec) (internal), this PR introduces "RUM View Ended" telemetry. The primary goal is to measure the quality of view timings (INV, TNS, and addViewLoadingTime()), laying the groundwork for overall view monitoring — similar to how ["RUM Session Ended"](https://github.com/DataDog/dd-sdk-ios/pull/1866) provides session insights.

#### TNS and INV Changes

First, this PR refactors `TNSMetric` and `INVMetric` to report their values through `Result<TimeInterval, NoValueReason>` rather than a simple `TimeInterval?`. This gives finer-grained insight into why a metric might be unavailable, using new enums `TNSNoValueReason` and `INVNoValueReason`:

```diff
// TNSMetric
- func value(with appStateHistory: AppStateHistory) -> TimeInterval?
+ func value(with appStateHistory: AppStateHistory) -> Result<TimeInterval, TNSNoValueReason>

// INVMetric
- func value(for viewID: RUMUUID) -> TimeInterval?
+ func value(for viewID: RUMUUID) -> Result<TimeInterval, INVNoValueReason>
```
To accommodate the range of "no value" reasons, the internal implementations of both metrics were refactored. `TNSMetric` in particular underwent more significant changes due to its broader set of edge cases — but thanks to good existing test coverage, this refactoring remains well-validated.

#### `ViewEndedMetric`

Second, this PR introduces "RUM View Ended" telemetry, which is sent whenever a view ends with **effective sample rate of 0.15%**.

- The design follows the pattern established in "RUM Session Ended".
- The metric tracks various view-level attributes, including newly captured TNS and INV values (or the corresponding "no value reasons").

```swift
internal final class ViewEndedMetricController {
    func track(viewEvent: RUMViewEvent, instrumentationType: ViewInstrumentationType?)
    func track(networkSettledResult: Result<TimeInterval, TNSNoValueReason>)
    func track(interactionToNextViewResult: Result<TimeInterval, INVNoValueReason>)
    func send()
}
```
```swift
class ViewEndedMetric {
    // ...

    /// Duration of the view in nanoseconds (equal to `@view.time_spent`).
    var durationNs: Int64?

    /// The value or "no value reason" for `@view.network_settled_time` (Time-to-Network-Settled, TNS).
    var tnsResult: Result<TimeInterval, TNSNoValueReason>?

    /// The value or "no value reason" for `@view.interaction_to_next_view_time` (Interaction-to-Next-View, INV).
    var invResult: Result<TimeInterval, INVNoValueReason>?

    // ...
}
```

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes
- [ ] Add Objective-C interface for public APIs (see our [guidelines](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/3157787243/RFC+-+Modular+Objective-C+Interface#Recommended-solution) [internal]) and run `make api-surface`)

### CODE REVIEW

File: Datadog/IntegrationUnitTests/RUM/SDKMetrics/RUMSessionEndedMetricIntegrationTests.swift [hunk 1]
```diff
@@ -262,14 +263,6 @@ class RUMSessionEndedMetricIntegrationTests: XCTestCase {
     }
 }
 
-private extension RUMSessionEndedMetricIntegrationTests {
-    func rumMonitor(with sessionEndedSampleRate: SampleRate = .maxSampleRate) -> RUMMonitorProtocol {
-        let monitor = RUMMonitor.shared(in: core)
-        monitor.dd.scopes.dependencies.sessionEndedMetric.sampleRate = sessionEndedSampleRate
-        return monitor
-    }
-}
```
Comment (ncreated) [inline Datadog/IntegrationUnitTests/RUM/SDKMetrics/RUMSessionEndedMetricIntegrationTests.swift:271]: cleanup/ This was hooking too much into implementation details, not good for integration testing. Replaced with passing the value from configuration:
```swift
rumConfig.sessionEndedMetricSampleRate = .maxSampleRate
```

File: DatadogRUM/Sources/RUMMetrics/TNSMetric.swift [hunk 2]
```diff
@@ -36,150 +36,226 @@ internal protocol TNSMetricTracking {
     /// Returns the value for the TNS metric.
     ///
     /// - Parameters:
-    ///   - time: The current time (device time, no NTP offset).
     ///   - appStateHistory: The history of app state transitions.
     /// - Returns: The value for the TNS metric, or `nil` if the metric cannot be calculated.
-    func value(at time: Date, appStateHistory: AppStateHistory) -> TimeInterval?
```
Comment (ncreated) [inline DatadogRUM/Sources/RUMMetrics/TNSMetric.swift:42]: 💡 The `time:` parameter was not used at all, it was a leftover from previous work.

**Review by simaoseica-dd — APPROVED** [review]
Great job. Thumbs up 👍 for returning a `Result` of the metrics instead of optionals that miss on adding details.

File: DatadogRUM/Sources/RUMConfiguration.swift [hunk 3]
```diff
@@ -289,6 +289,10 @@ extension RUM {
 
         /// An extra sampling rate for configuration telemetry events. It is applied on top of the value configured in public `telemetrySampleRate`.
         internal var configurationTelemetrySampleRate: SampleRate = 20.0
+        /// Sample rate for "view ended" metric in telemetry.
+        internal var viewEndedMetricSampleRate = ViewEndedMetricController.defaultSampleRate
+        /// Sample rate for "session ended" metric in telemetry.
+        internal var sessionEndedMetricSampleRate = SessionEndedMetricController.defaultSampleRate
```
[the remark this answers is not in the document]
Comment (ncreated) [reply DatadogRUM/Sources/RUMConfiguration.swift]: That makes sense 👍, done ✅

**Review by simaoseica-dd — APPROVED** [review]
LGTM! 👌
